CommunicationLayer/MessagingFramework/MotorDetailsMessage.cpp

```diff
@@ -0,0 +1,153 @@
+    const int VOLTAGE_RAIL_SUPPLY_19_V = 41;
+    const int HEAT_SINK_TEMPERATURE = 45;
+    const int MOTOR_TEMPTERATURE = 49;
+    const int DSP_BOARD_TEMPERATURE = 53;
+    const int DC_BUS_AMP_HOURS = 57;
+    const int ODOMETER = 61;
+    const int SLIP_SPEED = 65;
+}
+
+MotorDetailsMessage::MotorDetailsMessage(const QByteArray& messageData, unsigned char motorNumber)
+: messageData_(messageData)
+, motorNumber_(motorNumber)
+{
+}
+
+float MotorDetailsMessage::phaseCCurrent() const
+{
+    return getFloat(messageData_, PHASE_CCURRENT);
+}
+
+float MotorDetailsMessage::phaseBCurrent() const
+{
+    return getFloat(messageData_, PHASE_BCURRENT);
+}
+
+float MotorDetailsMessage::motorVoltageReal() const
+{
+    return getFloat(messageData_, MOTOR_VOLTAGE_REAL);
+}
+
+float MotorDetailsMessage::motorVoltageImaginary() const
+{
+    return getFloat(messageData_, MOTOR_VOLTAGE_IMAGINARY);
+}
+
+float MotorDetailsMessage::motorCurrentReal() const
+{
+    return getFloat(messageData_, MOTOR_CURRENT_REAL);
+}
+
+float MotorDetailsMessage::motorCurrentImaginary() const
+{
+    return getFloat(messageData_, MOTOR_CURRENT_IMAGINARY);
+}
+
+float MotorDetailsMessage::backEmfReal() const
+{
+    return getFloat(messageData_, BACK_EMF_REAL);
+}
+
+float MotorDetailsMessage::backEmfImaginary() const
+{
+    return getFloat(messageData_, BACK_EMF_IMAGINARY);
+}
+
+float MotorDetailsMessage::voltageRailSuppply15V() const
+{
+    return getFloat(messageData_, VOLTAGE_RAIL_SUPPPLY_15_V);
+}
+
+float MotorDetailsMessage::voltageRailSupply33V() const
+{
+    return getFloat(messageData_, VOLTAGE_RAIL_SUPPLY_33_V);
+}
+
+float MotorDetailsMessage::voltageRailSupply19V() const
+{
+    return getFloat(messageData_, VOLTAGE_RAIL_SUPPLY_19_V);
+}
+
+float MotorDetailsMessage::heatSinkTemperature() const
+{
+    return getFloat(messageData_, HEAT_SINK_TEMPERATURE);
+}
+
+float MotorDetailsMessage::motorTempterature() const
+{
+    return getFloat(messageData_, MOTOR_TEMPTERATURE);
+}
+
+float MotorDetailsMessage::dspBoardTemperature() const
+{
+    return getFloat(messageData_, DSP_BOARD_TEMPERATURE);
+}
+
+float MotorDetailsMessage::dcBusAmpHours() const
+{
+    return getFloat(messageData_, DC_BUS_AMP_HOURS);
+}
+
+float MotorDetailsMessage::odometer() const
+{
+    return getFloat(messageData_, ODOMETER);
+}
+
+float MotorDetailsMessage::slipSpeed() const
+{
+    return getFloat(messageData_, SLIP_SPEED);
+}
+
+QString MotorDetailsMessage::toString() const
+{
+    QString messageString;
+    if (motorNumber_ == 0x0) {
+        messageString += QString::number(MessageDefines::Motor0Details) + ", ";
+    }
+    else {
+        // For the sake of not halting programs execution, treat all other cases as motor one
+        if (motorNumber != 0x1) {
+            // Log that this was used incorrectly
+            qDebug() << "Invalid motor number, defaulting to motor 1";
+        }
+        messageString += QString::number(MessageDefines::Motor1Details) + ", ";
+    }
```

Comment:
@enochtsang 

Ok I'm assuming this is the thread you were confused about. so currently my implementation is 
```
if ((motorNumber & 0xe) != 0x0) {
        qDebug() << "Invalid motor number, unexpected behaviour likely";
    }
```
Which, assuming my logic is correct, will output a debug statement if the program receives a motor number other than 0 or 1. I understand now that it should not halt the execution because this is something being sent to Hermes and is not internal to the programs execution, so there is really no way to guarantee this situation doesn't occur. If this is an appropriate way of dealing with the problem let me know (All I'm doing is outputting this debug message before continuing with the normal flow of the program), or if there is something you have in mind that's better I'd be happy to alter this. :)